### PR TITLE
Update README with object_qsort

### PR DESCRIPTION
**Pull request by r-devulap**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -95,6 +142,33 @@ argselect) will not use the SIMD based algorithms if they detect NAN's in the
 array. You can read details of all the implementations
 [here](https://github.com/intel/x86-simd-sort/blob/main/src/README.md).
 
+## Performance comparison on AVX-512: `object_qsort` v/s `std::sort`
+Performance of `object_qsort` can vary significantly depending on the defintion
+of the custom class and we highly recommend benchmarking before using it. For
+the sake of illustration, we provide a few examples in
+[./benchmarks/bench-objsort.hpp](./benchmarks/bench-objsort.hpp) which measures
+performance of `object_qsort` relative to `std::sort` when sorting an array of
+points in the cartesian coordinates represented by the class: `struct Point
```
Comment (sterrettm2) [inline README.md:151]: I think I might rephrase this, I think it's a bit unnecessarily specific - I think I would just call them something like "an array of 3D points represented by"

File: README.md [hunk 2]
```diff
@@ -95,6 +142,33 @@ argselect) will not use the SIMD based algorithms if they detect NAN's in the
...
+{double x, y, z;}` and `struct Point {float x, y, x;}`. We sort these points
+based on several different metrics:
+
++ sort by coordinate `x`
++ sort by manhanttan distance (relative to origin): `abs(x) + abx(y) + abs(z)`
++ sort by Euclidean distance (relative to origin): `sqrt(x*x + y*y + z*z)`
++ sort by Chebyshev distance (relative to origin): `max(x, y, z)`
```
Comment (sterrettm2) [inline README.md:158]: I think the formula for Chebyshev distance should technically be max(abs(x), abs(y), abs(z)),

File: README.md [hunk 3]
```diff
@@ -1,32 +1,57 @@
...
+element in the array of objects of type `T`. `Func` is a lambda function that
+computes the `key` value for each object which is the metric used to sort the
+objects. `Func` needs to have the following signature:
 
+```cpp
+[] (T obj) -> type_t { type_t key; /* compute key for obj */ return key; }
+```
 
-### Sort routines on arrays
+Note that the return type of the key `type_t` needs to be one of the following
```
Comment (sterrettm2) [inline README.md:26]: I think I would rename the type used for the key to something like key_t; I think it risks causing confusion, like potentially misreading the next lines about the permitted types and space complexity

**Review by sterrettm2 — COMMENTED** [review]
Looks great overall, there were just a few things I noticed that might warrant changing

File: README.md (hunk 2 above)
Comment (r-devulap) [reply README.md]: that is correct :) fixed it.

**Review by sterrettm2 — APPROVED** [review]
Looks good to me!

**Pull request merged**